.travis.yml

```diff
@@ -30,13 +30,14 @@ matrix:
           SETUPOPTS=""
 
 install:
-  - sudo pip install -r requirements.txt
-  - sudo pip install -r test-requirements.txt
+  - pip install -r requirements.txt
+  - pip install -r test-requirements.txt
 
 script:
   # future: - pylint setup.py makepkg.py azurelinuxagent/
-  - sudo env "PATH=$PATH" nosetests $NOSEOPTS tests
-  - if [ ! -z "$SETUPOPTS" ]; then sudo /usr/bin/env python setup.py $SETUPOPTS; fi
+  - nosetests $NOSEOPTS tests
+  - sudo env "PATH=$PATH" nosetests $NOSEOPTS --verbose -a permissions=root tests
```

Comment:
How does Travis detect test failures? If any of the commands in the "script:" fails will it report a failure?

Reply:
Yes, it outputs the exit code of each line in the 'script' section, e.g.:
`The command "sudo env "PATH=$PATH" nosetests $NOSEOPTS --verbose -a permissions=root tests" exited with 0.`

Reply:
ok

Reply:
Discussed offline, the 'script' lines are executed sequentially and are not dependent on each other.